src/AutoRest.CSharp/Common/Output/Models/ClientFields.cs

```diff
@@ -132,6 +133,9 @@ private ClientFields(IEnumerable<Parameter> parameters, InputAuth? authorization
         public FieldDeclaration? GetFieldByParameter(Parameter parameter)
             => GetFieldByParameter(parameter.Name, parameter.Type);
 
+        public FieldDeclaration? GetFieldByName(string name)
```

Comment:
Provide this method for cases when we don't check type matching. I cannot remove `GetFieldByPrarameter`, since in some cases we still need to check the type. For example, `NamesConflict` test project provide a case when client parameter and query parameter have the same name. We actually rely on the type check to decide which should be used. Not sure if that's the best way to do.